server/src/actors/events.rs

```diff
@@ -550,6 +559,48 @@ impl Handler<HandleEvent> for EventManager {
             .map_err(move |error, _, _| {
                 log::warn!("error processing event {}: {}", event_id, LogError(&error));
                 metric!(counter("event.rejected") += 1);
+                let outcome_params: Option<(Outcome, OutcomeReason)> = match &error {
+                    ProcessingError::InvalidJson(_)
```

Comment:
This is kind-of special. It's not an internal failure if the JSON payload from a client is invalid. This should have an `InvalidJson` reason, similar to when the payload is too large.

Reply:
Ok, done